### DISCUSSION TITLE
Nulleinspeisung mit Shelly 3EM realisierbar ?

### DISCUSSION THREAD

**Discussion by LJ9210** [tbnobody/OpenDTU#209] [Q&A]
ist eine Nulleinspeisung mit einem Shelly 3EM machbar ? 

bzw. welche Möglichkeiten gibt es noch um eine Nulleinspeisung mit Open DTU zu realisieren ?

**Comment by helgeerbe** [comment]
Ist sicher möglich für ein einfaches Grundlastkraftwerk.
Ich selbst habe den 3EM als Monitor.
Mein Batteriekraftwerk (für die Grundlast) wird über einen Victron Batterielader geladen.
Auf AC Seite schaltet ein Schelly 1PM den Inverter aus, wenn Energie ins Netz abließt (dann lieber die Batterie weiter laden). Aktuell ist mein Inverter fix auf 150W eingestellt. Ich will ihn zukünftig durch einen HM-300 ersetzen, den ich dann hoffentlich dynamisch regeln kann.

Ob sich das finanziell rechnet steht auf einem ganz anderen Blatt. Aber beim Hobby fragt ja auch keiner nach dem Geld.
Die aktuell Lösung habe ich im Laufe des Julis in Betrieb genommen. Bisher sind ca. 20 kWh (ca. 7.50 €) abgeflossen. Ein HM-300 kostest so um die 150€.......

**Reply by stromibaer** [reply]
Was hat der 3EM für ein Image laufen? Ist da Tasmota drauf oder ...? 
Suche einen günstigen Stromzähler mit Stromwandler, also den Ringen und der MqTT spricht.

**Reply by daebikelinz** [reply]
@helgeerbe Haben sie die Akkus direkt mit dem Inverter verbunden? 
Welchen Inverter verwenden sie? 
Haben sie DC hart verbunden und keine Probleme feststellen können?

**Reply by madmartin** [reply]
Hallo @helgeerbe ,
mich interessiert ebenfalls Deine Geamt-Konfiguration, und Deine Entwicklung von https://github.com/helgeerbe/OpenDTU_VeDirect  hab ich auch entdeckt. Kannst Du vielleicht mal ein Gesamt-Bild  mit den Komponenten zeichnen? Und am besten bei https://github.com/helgeerbe/OpenDTU_VeDirect/docs ablegen?

**Reply by helgeerbe** [reply]
@daebikelinz 
Sorry, für die späte Antwort. 
Aktuell verwende ich für mein Batteriekraftwerk einen billigen Inverter von [Ali Express](https://www.aliexpress.com/item/1005003338787245.html?_randl_currency=EUR&_randl_shipto=DE&src=google&memo1=freelisting&src=google&albch=shopping&acnt=494-037-6276&slnk=&plac=&mtctp=&albbt=Google_7_shopping&albagn=888888&isSmbAutoCall=false&needSmbHouyi=false&albcp=14917676811&albag=127853744826&trgt=449349191507&crea=de1005003338787245&netw=u&device=c&albpg=449349191507&albpd=de1005003338787245&gclid=Cj0KCQjwteOaBhDuARIsADBqRejQ5ugbX7Ji1H9WeGWfRWT-Vw-2fn8Y97uCT7NZUBdsVDAeSUFF5AQaArOGEALw_wcB&gclsrc=aw.ds&aff_fcid=581359fa17f34f31bb7eeea42a357a48-1666771600283-05357-UneMJZVf&aff_fsk=UneMJZVf&aff_platform=aaf&sk=UneMJZVf&aff_trace_key=581359fa17f34f31bb7eeea42a357a48-1666771600283-05357-UneMJZVf&terminal_id=8362d18abd784c39b7ed5e94c6979d99&afSmartRedirect=y) 

Dort hängt die 24V Batterie direkt dran. Die Leistung wird über ein Poti gesteuert (fest eingestellt) und über einen Shelly 1 kann ich den Inverter nach Bedarf an und aus schalten. Funktioniert ganz gut. Ausser, dass immer eine feste Leistung eingespeist wird. 

Hier hätte ich die Hoffnung, dass man über einen HM-300 die Leistung dynamisch an den tatsächlichen Bedarf anpassen kann. Wenn ich das Datenblatt richtig verstanden habe, sollte das funktionieren. Getestet habe ich es allerdings noch nicht. Ich habe noch einen HM-600 in Betrieb und muss mal bei Gelegenheit die Kabel tauschen und das testen. Falls jemand schon Erfahrung mit einer Batterie am HM-300 hat, würde mich das natürlich interessieren.

**Reply by helgeerbe** [reply]
Hallo @madmartin,
ich hab's mal auf die Schnelle versucht.
![diagram](https://github.com/helgeerbe/OpenDTU_VeDirect/blob/master/docs/components.png)

**Comment by helgeerbe** [comment]
Das bringt mich jetzt zu einer weiteren Frage. Über MQTT kann man den Inverter ausschalten. Was wird tatsächlich ausgeschaltet, bzw. wie hoch ist die Leistungsaufnahme im Aus-Zustand, wenn eine Batterie daran hängt. Momentan schalte ich den Inverter hart über einen Schelly am AC Ende aus. Findet da tatsächlich eine Netztrennung statt? Oder sollte man lieber weiterhin das Relais des Schelly verwenden?

**Reply by dsteinkopf** [reply]
Meine Beobachtung: Bei via MQTT ausgeschaltetem Inverter zeigt der Shelly 0W an. Dort findet also kein Verbrauch statt. Der Inverter selbst "sagt", er verbrauche 2-3 Watt DC-seitig (hab die Zahl nicht genau im Kopf). Den gleichen DC-Verbrauch hat er aber auch, wenn man die Netzverbindung im Shelly trennt. Das macht also keinen Unterschied.

Der größte Unterschied, den ich beobachte, ist die lange "Anlaufphase" des Inverters, wenn er vom Netz getrennt war: Er muss sich offenbar erst lange mit dem Netz synchronisieren und fährt danach die Inverterleistung langsam bis zu gewünschten Wert hoch. Das dauert mehrere Minuten. Änderungen der Leistung, sowie Ein-/Ausschalten via MQTT geht deutlich schneller: Das sind wenige Sekunden bis ca. 30 Sekunden.

Also schalte ich die Leistung immer nur via MQTT an/aus/hoch/runter. Manchmal "verstolpert" sich aber "irgendwas", dann bleibt die Leistung konstant - egal was ich sende. Ahoy scheint aber noch mit dem Inverter zu "sprechen" - hat aber keine Wirkung. DANN trenne ich (manuell) via Shelly vom Netz und wieder an. Dann klappt es meistens wieder.

**Reply by dsteinkopf** [reply]
Zur Sicherheit (weil ich's übersehen hatte). Ich spreche von **Ahoy**. Wir sind hier ja bei OpenDTU. Sorry.

**Reply by FrodoVDR** [reply]
Lohnt sich das abschalten überhaupt? Mein Shelly zeigt 0 A an wenn die Sonne nichts mehr produziert. Laut dem Datenblatt sind es beim HM-600 < 50 mW .

**Reply by helgeerbe** [reply]
Na, wenn die Panels direkt daran hängen lohnt sich das Abschalten sicher nicht. Ich will aber über eine Batterie einspeisen und da will ich schon steuern ob und wie hoch die Batterie belastet wird. Wenn Leistung 0% und Abschaltung das gleiche bewirken brauche ich die Abschaltung tatsächlich nicht und würde nur über die Leistung steuern.

**Reply by kpwg** [reply]
Noch ein Hinweis zu MQTT generell: Das Protokoll bedingt IMMER einen Broker, welcher dann die Rechnerei auch mit übernehmen könnte. Eine Kommunikation von Client zu Client ist bei MQTT nicht vorgesehen, somit gibt es auch keine direkte Kommunikation von z.B. Shelly 3EM zu OpenDTU.

Ich schalte den Inverter (1200er, ein Kanal in Benutzung) nur über MQTT an und aus, AC-seitig bleibt der stets online. Dabei habe ich einen Verbrauch von 3-4W, wobei die Batterie auch einen Victron 24/375 dauerhaft im eco Modus betreibt. Dafür könnte ich in wenigen Sekunden auf sich ändernde Lasten reagieren und zuschalten. Die AC-seitige Trennung sollte immer nur zweite Wahl sein, denn das Synchronisieren mit dem Netz kostet viel Zeit.

**Reply by stefan123t** [reply]
@kpwg aber sowohl Shelly als auch OpenDTU können sich auch gegenseitig ihre Topics auf dem MQTT Broker abonnieren. D.h. man könnte OpenDTU durchaus beibringen einzelne Topics einer Shelly 3EM zu lesen, selbst aufzusummieren / saldieren und dann die WR Leistung selbst anzupassen. Dann brauche ich an der Stelle keine andere Home Automatisierung.

**Reply by madmartin** [reply]
@stefan123t Du brauchst aber immer - wie  @kpwg richtig bemerkt - einen MQTT Broker. Und auf dem Rechner, wo dieser Broker arbeitet, könnten dann auch diese Berechnungen und Steuerungen stattfinden.
Meiner Meinung nach gehört sowas nicht in eine Daten-Bridge wie OpenDTU hinein.

**Comment by daebikelinz** [comment]
Ja das ist möglich mit dem Shelly 3EM. Habe ich selbst gerade am Laufen und zwar so:

Eine OpenHAB Rule setzt bei jedem neuen Leistungswert des Shelly 3EM ein neues Einspeiselimit des TSUN M350 per MQTT Topic "SolarGateway/11217011xxxx/cmd/limit_nonpersistent_relative" über das openDTU Gateway. 
Von 12:05 Uhr bis 12:25 Uhr war kurz mal zu wenig Sonne um das eingestellte Limit auch einzuspeisen. 
Zielwert auf den geregelt werden soll, waren -5W Netzeinspeisung für das Bild unten.

Nächster Schritt: Akku statt PV als Input! 
Macht aber jetzt schon richtig Laune das in Aktion zu sehen.

![M350_Limit_1](https://user-images.githubusercontent.com/39101639/200196734-8b346124-6c4a-4f56-aec5-788a6b405676.jpg)

**Reply by LJ9210** [reply]
@daebikelinz Wie kann ich mir so eine openHAB Regel vorstellen ? Läuft das über eine feste Automation oder werden mehrere Automationen benötigt ? 

Hintergrund meiner frage ist das ich per Home Assistant versuche es zu realisieren aber noch nicht so genau weiß welche Auslöser/Bedingungen/Aktionen zusammen spielen müssen :)

**Reply by daebikelinz** [reply]
Ich weiß nicht genau was mit Automation gemeint ist und habe keine Erfahrung mit Home Assistant
Die OpenHAB Regel sieht wie folgt aus:

```
rule "Adjust inverter limit"
when
    Item P_EM3_residual received update
then
    var offset = 0 // 0 Watt target to regulate on!
    var Number newlimit = (M350_power.state as DecimalType + (P_EM3_residual.state as DecimalType - offset)) /3 // * 100 % / 300 W = 1/3
    if (newlimit >= 100)   newlimit = 100
    if (newlimit <= 0)    newlimit = 0
    var Number diff_PtoLimit = M350_power.state as DecimalType - (M350_limit_rel.state as DecimalType * 3)
    postUpdate(diff_Power_to_Limit, diff_PtoLimit) 

    if (((newlimit < 100) || (M350_limit_rel.state != 100)) && (P_EM3_residual.state < 10 || ((diff_PtoLimit <= 5) && (diff_PtoLimit >= -5)))) {
            
        var Number diff_r = newlimit - M350_limit_rel.state as DecimalType
        if ((diff_r >= 1.5) || (diff_r <= -1.5)) {
            postUpdate(limit_nonpersistent_relative, newlimit)
        }
    }
end
```

**Reply by LJ9210** [reply]
![8FA662D2-00A8-41E6-A6BF-ACF62C4FCF37](https://user-images.githubusercontent.com/75933247/201899920-f7bfed1c-43ba-436f-8d1e-365069d737af.png)

@daebikelinz habe jemanden gefunden der openHAB rules in Home Assistant Templates umschreiben kann. 

Kannst du zu seinen fragen was sagen ? Ich persönlich wäre raus habe dafür zu wenig Ahnung 😬

**Reply by daebikelinz** [reply]
Ich denke die zwei Systeme sind wohl grundsätzlich bisschen unterschiedlich nach dem was ich lese.
Alles nach "//" ist ein Kommentar und kann vernachlässigt werden. Dadurch soll man den Code lediglich besser verstehen können.
M350_Power ist die Leistung des Balkonkraftwerks
P_EM3_residual ist die Leistung der Addition der drei Phasen des Shelly EM3
M350_limit_rel ist das relative Limit des TSUN M350 Wechselrichters was er auf MQTT ausgibt.

Da wird nur die Variable geupdatet. Ist mir einem "=" gleichzusetzen.
Erste nur für debug Zwecke
Zweites schreibt das neue Limit welches dann auf MQTT gepublished wird.
